protocol/directory.go

```diff
@@ -39,18 +65,43 @@ func (d *ConiksDirectory) Update() {
 	}
 }
 
+// SetPolicies sets this ConiksDirectory's epoch deadline and VRF
+// private key.
 func (d *ConiksDirectory) SetPolicies(epDeadline merkletree.TimeStamp, vrfKey vrf.PrivateKey) {
 	d.policies = merkletree.NewPolicies(epDeadline, vrfKey)
 }
 
+// EpochDeadline returns this ConiksDirectory's current epoch deadline
+// as a timestamp.
 func (d *ConiksDirectory) EpochDeadline() merkletree.TimeStamp {
 	return d.pad.LatestSTR().Policies.EpochDeadline
 }
 
+// LatestSTR Returns this ConiksDirectory's latest STR.
 func (d *ConiksDirectory) LatestSTR() *merkletree.SignedTreeRoot {
 	return d.pad.LatestSTR()
 }
 
+// Register inserts the username-to-key mapping contained in a
+// RegistrationRequest req received from a CONIKS client
+// into this ConiksDirectory, and returns a tuple of the form
+// (response, error).
+// The response (which also includes the error code) is supposed to
+// be sent back to the client. The returned error is used by the key
+// server for logging purposes.
+//
+// A request without a username or without a public key is considered
+// malformed, and causes Register() to return a (error response,
+// ErrorMalformedClientMessage) tuple.
+// If the given username already exists in the latest snapshot of the
+// directory, Register() returns an (error response, ErrorNameExisted)
+// tuple.
+// Otherwise, Register() inserts the new mapping in req
+// into the PAD so it can be included in the snapshot taken at the end
+// of the current  epoch), and returns a (registration proof, Success)
+// if this operation succeeds.
+// If Register() encounters an internal error at any point, it returns
+// an (error response, ErrorDirectory) tuple.
```

Comment:
No, it does assume that `useTBs` is true. I've added in the TB cases to make this explanation more complete.